### ISSUE TITLE
npm install failing on prepublish sk-bundle ( windows only )

### ISSUE THREAD

**Issue by kentendo** [skatejs/skatejs#961]
I tried this on multiple windows boxes with different versions of npm and node. Works great on Mac :)

windows ( 7 - 10 )
node ( 4.2.6 - 6.9.1 ) 
npm ( 2.14.12 - 3.10.8 )

1. git clone https://github.com/skatejs/skatejs.git
2. npm install
3. fail

```
The system cannot find the path specified.

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "prepublish"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! skatejs@ prepublish: `sk-bundle`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the skatejs@ prepublish script 'sk-bundle'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the skatejs package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     sk-bundle
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs skatejs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls skatejs
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Git\skatejs\npm-debug.log

```

**Comment by treshugart** [comment]
Could you try and find the relevant error in `npm-debug.log` and post that, please? Thanks.

**Comment by kentendo** [comment]
Reading this, the error is still not clear to me, but here are the relevant logs.
```
52 silly lifecycle skatejs@~preinstall: no script for preinstall, continuing
53 silly executeActions Starting
54 silly install executeActions
55 silly doSerial global-install 0
56 silly doParallel fetch 0
57 verbose correctMkdir C:\Users\kcosca\AppData\Roaming\npm-cache\_locks correctMkdir not in flight; initializing
58 verbose lock using C:\Users\kcosca\AppData\Roaming\npm-cache\_locks\staging-ff118b6297fd35a4.lock for C:\Git\skatejs\node_modules\.staging
59 silly doParallel extract 0
60 silly doParallel preinstall 0
61 silly doReverseSerial remove 0
62 silly doSerial move 0
63 silly doSerial finalize 0
64 silly doSerial build 0
65 silly doSerial global-link 0
66 silly doParallel update-linked 0
67 silly doSerial install 0
68 silly doSerial postinstall 0
69 verbose unlock done using C:\Users\kcosca\AppData\Roaming\npm-cache\_locks\staging-ff118b6297fd35a4.lock for C:\Git\skatejs\node_modules\.staging
70 silly executeActions Finishing
71 silly rollbackFailedOptional Starting
72 silly rollbackFailedOptional Finishing
73 silly install runPostinstallTopLevelLifecycles
74 silly build skatejs@
75 info linkStuff skatejs@
76 silly linkStuff skatejs@ has C:\Git as its parent node_modules
77 verbose linkBins skatejs@
78 verbose linkMans skatejs@
79 silly install skatejs@ C:\Git\skatejs\skatejs-0363025d
80 info lifecycle skatejs@~install: skatejs@
81 silly lifecycle skatejs@~install: no script for install, continuing
82 silly postinstall skatejs@ C:\Git\skatejs\skatejs-0363025d
83 info lifecycle skatejs@~postinstall: skatejs@
84 silly lifecycle skatejs@~postinstall: no script for postinstall, continuing
85 silly prepublish skatejs@ C:\Git\skatejs\skatejs-0363025d
86 info lifecycle skatejs@~prepublish: skatejs@
87 verbose lifecycle skatejs@~prepublish: unsafe-perm in lifecycle true
88 verbose lifecycle skatejs@~prepublish: PATH: C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin;C:\Git\skatejs\node_modules\.bin;P:\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\local\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\usr\bin;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;P:\bin;C:\ProgramData\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit;C:\Program Files\Microsoft SQL Server\110\Tools\Binn;C:\Program Files (x86)\AMD\ATI.ACE\Core-Static;C:\Program Files (x86)\Card Scanning Solutions\SDK;C:\Program Files\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn;C:\Program Files (x86)\Microsoft SQL Server\110\Tools\Binn\ManagementStudio;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Common7\IDE\PrivateAssemblies;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files (x86)\Microsoft SDKs\TypeScript\1.0;C:\Program Files\Microsoft SQL Server\120\Tools\Binn;C:\Program Files\nodejs;C:\Users\kcosca\AppData\Roaming\npm;C:\Program Files\Git\usr\bin\vendor_perl;C:\Program Files\Git\usr\bin\core_perl
89 verbose lifecycle skatejs@~prepublish: CWD: C:\Git\skatejs
90 silly lifecycle skatejs@~prepublish: Args: [ '/d /s /c', 'sk-bundle' ]
91 silly lifecycle skatejs@~prepublish: Returned: code: 1  signal: null
92 info lifecycle skatejs@~prepublish: Failed to exec prepublish script
93 warn optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
94 warn notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.0.15: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
95 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid OS:    darwin
95 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Valid Arch:  any
95 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual OS:   win32
95 verbose notsup SKIPPING OPTIONAL DEPENDENCY: Actual Arch: x64
96 verbose stack Error: skatejs@ prepublish: `sk-bundle`
96 verbose stack Exit status 1
96 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:255:16)
96 verbose stack     at emitTwo (events.js:106:13)
96 verbose stack     at EventEmitter.emit (events.js:191:7)
96 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:40:14)
96 verbose stack     at emitTwo (events.js:106:13)
96 verbose stack     at ChildProcess.emit (events.js:191:7)
96 verbose stack     at maybeClose (internal/child_process.js:877:16)
96 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
97 verbose pkgid skatejs@
98 verbose cwd C:\Git\skatejs
99 error Windows_NT 6.1.7601
100 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "--loglevel" "verbose"
101 error node v6.9.1
102 error npm  v3.10.8
103 error code ELIFECYCLE
104 error skatejs@ prepublish: `sk-bundle`
104 error Exit status 1
105 error Failed at the skatejs@ prepublish script 'sk-bundle'.
105 error Make sure you have the latest version of node.js and npm installed.
105 error If you do, this is most likely a problem with the skatejs package,
105 error not with npm itself.
105 error Tell the author that this fails on your system:
105 error     sk-bundle
105 error You can get information on how to open an issue for this project with:
105 error     npm bugs skatejs
105 error Or if that isn't available, you can get their info via:
105 error     npm owner ls skatejs
105 error There is likely additional logging output above.
106 verbose exit [ 1, true ]

```

**Comment by treshugart** [comment]
Seems the issue happens in https://github.com/skatejs/build/blob/master/bin/sk-bundle.

**Comment by andreawyss** [comment]
It is caused by fsevents. 
Unsupported platform for fsevents@1.0.15: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

**Comment by treshugart** [comment]
It seems you might be able to [disable optional deps](https://github.com/npm/npm/issues/9204#issuecomment-216853959) and it might work?

**Comment by Qiu-Kejian** [comment]
@andreawyss 
Is this caused by bash script in skatejs-build/bin/sk-bundle
`
#!/bin/sh

rm -rf dist/

if [ -f ./rollup.config.js ];
then
  ./node_modules/rollup/bin/rollup --config rollup.config.js "$@"
  ./node_modules/rollup/bin/rollup --config rollup.config.js --min "$@"
fi

if [ -f ./webpack.config.js ];
then
  ./node_modules/webpack/bin/webpack.js --config webpack.config.js "$@"
fi
`

# On the other hand, I tried run "rollup -c" from cmd line directly. Still failed and the error is a litter different:
Error transforming C:\projects\github\skatejs\src\index.js with 'babel' plugin: It looks like your Babel configuration s
pecifies a module transformer. Please disable it. If you're using the "es2015" preset, consider using "es2015-rollup" in
stead. See https://github.com/rollup/rollup-plugin-babel#configuring-babel for more information
Error: Error transforming C:\projects\github\skatejs\src\index.js with 'babel' plugin: It looks like your Babel configur
ation specifies a module transformer. Please disable it. If you're using the "es2015" preset, consider using "es2015-rol
lup" instead. See https://github.com/rollup/rollup-plugin-babel#configuring-babel for more information
    at preflightCheck (C:\projects\github\skatejs\node_modules\rollup-plugin-babel\dist\rollup-plugin-babel.cjs.js:43:10
2)
    at Object.transform$1 [as transform] (C:\projects\github\skatejs\node_modules\rollup-plugin-babel\dist\rollup-plugin
-babel.cjs.js:104:18)
    at C:\Users\PrimeVita\AppData\Roaming\npm\node_modules\rollup\src\utils\transform.js:19:35
    at process._tickCallback (internal/process/next_tick.js:103:7)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
Type rollup --help for help, or visit https://github.com/rollup/rollup/wiki
PS C:\projects\github\skatejs> webpack

**Comment by treshugart** [comment]
Is there any way we can fix this? It'd be worth testing after the migration to webpack 2 in master.

**Comment by treshugart** [comment]
Might be fixed in #1075.

**Comment by kentendo** [comment]
I can confirm this being fixed on windows 10 / node 6.10.2

**Comment by Hotell** [comment]
great! Closing. thx!
